test/integration/backward_compatible/parallel/topic/ReliableTopicTest.js

```diff
@@ -123,8 +143,35 @@ describe('ReliableTopicTest', function () {
         }).catch(done);
     });
 
+    it('removed message listener does not receive items after removal with new added methods', async function() {
+        const topicName = 't' + Math.random();
+        const topicOne = await clientOne.getReliableTopic(topicName);
+        const topicTwo = await clientTwo.getReliableTopic(topicName);
+        let receivedMessages = 0;
+        const callback = () => {
+            receivedMessages++;
+            if (receivedMessages > 2) {
+                throw new Error('Kept receiving messages after message listener is removed.');
+            }
+        };
+        const id = await topicTwo.addListener(callback);
+
+        topicOne.publish({ 'value0': 'foo0' });
+        topicOne.publish({ 'value1': 'foo1' });
+        setTimeout(() => {
+            topicTwo.removeListener(id);
+            topicOne.publish({ 'value2': 'foo2' });
+            topicOne.publish({ 'value3': 'foo3' });
+            topicOne.publish({ 'value4': 'foo4' });
+            topicOne.publish({ 'value5': 'foo5' });
+            setTimeout(() => {
+                // Test passes
+            }, 500);
+        }, 500);
```

Comment:
Instead of this you should do `await TestUtil.promiseWaitMilliseconds(500);`. With setTimeout, it is async and we don't wait for it.